_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,442 @@
+  type: ClusterIP
+  clusterIP: None
+  publishNotReadyAddresses: true
+  ports:
+    - name: peer
+      port: 27017
+  selector:
+    app: store-mongodb
+```
+
+Then deploy the application to Minikube, in the `kubernetes` directory, run:
+
+```bash
+./kubectl-apply.sh -f
+```
+Now is a good time to install [k9s](https://k9scli.io/topics/install/), a terminal-based UI to interact with Kubernetes clusters. Then run k9s with:
+
+```bash
+k9s -n demo
+```
+
+{% img blog/jhipster-digital-ocean/k9s-pods.png alt:"k9s UI" width:"800" %}{: .center-image }
+
+Check out the [commands list](https://k9scli.io/topics/commands/), some useful ones are:
+- `:namespace`: show all available namespaces
+- `:pods`: show all available pods
+You can navigate the pods with `ENTER` and go back with `ESC` keys.
+
+Set up port-forwarding for the JHipster Registry.
+
+```bash
+kubectl port-forward svc/jhipster-registry -n demo 8761
+```
+
+Navigate to `http://localhost:8761` and sign in with your Okta credentials. When the registry shows all services in green, Set up port-forwarding for the gateway as well.
+
+```bash
+kubectl port-forward svc/gateway -n demo 8080
+```
+
+Navigate to `http://localhost:8080`, sign in, and create some entities to verify everything is working fine.
+
+After looking around, stop minikube before the cloud deployment:
+
+```bash
+minikube stop
+```
+
+Also delete the minikube cluster to have a clean slate for future tests:
+
+```bash
+minikube delete
+```
+
+## Deploy to DigitalOcean Kubernetes
+
+Now that the architecture works locally, let's proceed to the cloud deployment. First, create a [DigitalOcean](https://www.digitalocean.com/try/free-trial-offer) account, you can try their services with a free $100 credit that is available for 60 days. The registration requires a $5 PayPal payment, or providing a credit card.
+
+Most of the cluster tasks, if not all, can be accomplished using [doctl](https://github.com/digitalocean/doctl#installing-doctl), the command-line interface (CLI) for the DigitalOcean API. Install the tool, and perform the authentication with DigitalOcean:
+
+```bash
+doctl auth init
+```
+You will be prompted to enter the DigitalOcean access token that you can generate in the DigitalOcean control panel. Sign in, and then in the left menu go to **API**, click **Generate New Token**. Enter a token name, and click **Generate Token**. Copy the new token from the **Tokens/Keys** table.
+
+{% img blog/jhipster-digital-ocean/do-control-panel-token.png alt:"DigitalOcean Control Panel for Token Create" width:"500" %}{: .center-image }
+
+You can find a detailed list of pricing for cluster resources at [DigitalOcean](https://docs.digitalocean.com/products/kubernetes/),  and with `doctl` you can quickly retrieve a list of node size options available for your account:
+
+```bash
+doctl k options sizes
+```
+
+Create the cluster with the following command line:
+
+```bash
+doctl k cluster create do1 -v --size s-4vcpu-8gb-intel
+```
```

Comment:
@indiepopart Do you see this option with the free trial? If not, maybe we should remove that signup link or ask DO to add this tier to the trial for JHipster. I don't think I used it and I've been billed about $8 so far.

Reply:
I had to submit a ticker to support to see the option s-4vcpu-8gb-intel, and they told me new users have limited node size, and they enabled all node sizes. I am not sure they can selectively enable node sizes depending on the framework. I will replace the free tial link back, set the standard registration link.

Reply:
Updated the registration link.